tests/platform_helper/domain/test_pipelines.py

```diff
@@ -0,0 +1,138 @@
+from pathlib import Path
+from unittest.mock import Mock
+from unittest.mock import patch
+
+import pytest
+import yaml
+from freezegun.api import freeze_time
+
+from dbt_platform_helper.constants import PLATFORM_CONFIG_FILE
+from dbt_platform_helper.domain.config_validator import ConfigValidator
+from dbt_platform_helper.domain.pipelines import Pipelines
+from dbt_platform_helper.providers.config import ConfigProvider
+from dbt_platform_helper.utils.messages import abort_with_error
+
+
+class PipelineMocks:
+    def __init__(self, app_name):
+        self.mock_config_provider = ConfigProvider(ConfigValidator())
+        self.mock_echo = Mock()
+        self.mock_abort = abort_with_error
+        self.mock_git_remote = Mock()
+        self.mock_git_remote.return_value = "uktrade/test-app-deploy"
+        self.mock_codestar = Mock()
+        self.mock_codestar.return_value = (
+            f"arn:aws:codestar-connections:eu-west-2:1234567:connection/{app_name}"
+        )
+
+    def params(self):
+        return {
+            "config_provider": self.mock_config_provider,
+            "echo": self.mock_echo,
+            "abort": self.mock_abort,
+            "get_git_remote": self.mock_git_remote,
+            "get_codestar_arn": self.mock_codestar,
+        }
+
+
+def test_pipeline_generate_with_empty_platform_config_yml_outputs_warning():
+    mock_config_provider = Mock()
+    app_name = "my-app"
+    mock_config_provider.load_and_validate_platform_config.return_value = {"application": app_name}
+
+    mocks = PipelineMocks(app_name)
+    mocks.mock_config_provider = mock_config_provider
+
+    pipelines = Pipelines(**mocks.params())
+    pipelines.generate(None, None)
+
+    mocks.mock_echo.assert_called_once_with(
+        "No pipelines defined: nothing to do.", err=True, fg="yellow"
+    )
+
+
+def test_pipeline_generate_with_non_empty_platform_config_but_no_pipelines_outputs_warning():
+    mock_config_provider = Mock()
+    mock_config_provider.load_and_validate_platform_config.return_value = {"environments": {}}
+
+    mocks = PipelineMocks("app-name")
+    mocks.mock_config_provider = mock_config_provider
+
+    pipelines = Pipelines(**mocks.params())
+    pipelines.generate(None, None)
+
+    mocks.mock_echo.assert_called_once_with(
+        "No pipelines defined: nothing to do.", err=True, fg="yellow"
+    )
+
+
+@freeze_time("2024-10-28 12:00:00")
+@patch("dbt_platform_helper.jinja2_tags.version", new=Mock(return_value="v0.1-TEST"))
```

Comment:
Haven't worked out a way of avoiding the jinja tag patching. Any suggestions welcome...